// Copyright 2021 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto2";

message SampleData {
  message SampleStruct {
    optional int64 sub_int_col = 1;
  }

  optional bool bool_col = 1;
  optional bytes bytes_col = 2;
  optional int32 date_col = 3;
  optional string datetime_col = 4;
  optional double float64_col = 5;
  optional string geography_col = 6;
  optional int64 int64_col = 7;
  optional string numeric_col = 8;
  optional string bignumeric_col = 9;
  optional string string_col = 10;
  optional string time_col = 11;
  optional int64 timestamp_col = 12;
  repeated int64 int64_list = 13;
  optional SampleStruct struct_col = 14;
  repeated SampleStruct struct_list = 15;
}
